### PR TITLE
Update nrfutil and dependencies.

### PR DESCRIPTION
**Pull request by KenMacD**
###### Description of changes

`nrfutil` and dependencies were updated to versions that include support for Python 3.10. The `broken` was removed by using the packaged version of `libusb1` instead of the one the project locked.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 180075` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nrfutil</li>
    <li>pc-ble-driver</li>
    <li>python310Packages.pc-ble-driver-py</li>
    <li>python39Packages.pc-ble-driver-py</li>
  </ul>
</details>

**Comment by jonas-schievink** [comment]
This looks reasonable to me, however I don't use NixOS anymore, so it would be great if someone could take over maintenance of `pc-ble-driver`.

File: pkgs/development/libraries/pc-ble-driver/default.nix [hunk 1]
```diff
@@ -15,6 +15,13 @@ stdenv.mkDerivation rec {
     sha256 = "1609x17sbfi668jfwyvnfk9z29w6cgzvgv67xcpvpx5jv0czpcdj";
   };
 
+  # Include upstream patch to allow building with GCC 11.
+  patches = (fetchpatch {
+    url = "https://github.com/NordicSemiconductor/pc-ble-driver/commit/37258e65bdbcd0b4369ae448faf650dd181816ec.patch";
+      sha256 = "sha256-gOdzIW8YJQC+PE4FJd644I1+I7CMcBY8wpF6g02eI5g=";
+    })
+  ;
```
Comment (AndersonTorres) [inline pkgs/development/libraries/pc-ble-driver/default.nix:23]:
```suggestion
  patches = [
     # Fix build with GCC 11
    (fetchpatch {
      url = "https://github.com/NordicSemiconductor/pc-ble-driver/commit/37258e65bdbcd0b4369ae448faf650dd181816ec.patch";
      sha256 = "sha256-gOdzIW8YJQC+PE4FJd644I1+I7CMcBY8wpF6g02eI5g=";
    })
  ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by KenMacD** [comment]
Applied suggested change, rebased to latest again, and bumped versions to latest versions again. Anything else needed?

**Comment by KenMacD** [comment]
@sbruder thank you, fixed the commit messages. Let me know if I should try to split the `pc-ble-driver-py` commit to have one with ` python310Packages.pc-ble-driver-py: init at 0.17.0`

**Comment by KenMacD** [comment]
Ah, that's how it's done. Makes sense. Updated and thanks again. I appreciate the help.

File: pkgs/development/libraries/pc-ble-driver/default.nix [hunk 2]
```diff
@@ -15,6 +15,14 @@ stdenv.mkDerivation rec {
     sha256 = "1609x17sbfi668jfwyvnfk9z29w6cgzvgv67xcpvpx5jv0czpcdj";
   };
 
+  patches = [
+     # Fix build with GCC 11
+    (fetchpatch {
+      url = "https://github.com/NordicSemiconductor/pc-ble-driver/commit/37258e65bdbcd0b4369ae448faf650dd181816ec.patch";
```
Comment (risicle) [inline pkgs/development/libraries/pc-ble-driver/default.nix:21]:
```suggestion
      name = "gcc11-fix.patch";
      url = "https://github.com/NordicSemiconductor/pc-ble-driver/commit/37258e65bdbcd0b4369ae448faf650dd181816ec.patch";
```

Comment (risicle) [reply pkgs/development/libraries/pc-ble-driver/default.nix:21]: Actually nevermind, let's get this moving..

**Review by risicle — APPROVED** [review]

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64

**Pull request merged**